src/backends/simple_expr.h

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+
+Author: Jared Roesch
+*/
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include "kernel/environment.h"
+
+
+namespace lean  {
+    using std::shared_ptr;
+
+    enum simple_expr_kind {
+        SVar,
+        Let,
+        Call,
+        Switch,
+        Error,
+        Project,
+        ClosureAlloc,
+    };
+
+    struct simple_expr {
```

Comment:
I didn't implement the intrusive smart pointer here out of a desire to more quickly, if you think we should do that let me know.